lib/utils/jsonPathUtils.ts

```diff
@@ -90,4 +90,45 @@ export class JsonPathUtils {
       }
     }
   }
+
+  static changeSpecialPathsRecursively(pd: IPresentationDefinition) {
+    const paths: { value: unknown; path: (string | number)[] }[] = JsonPathUtils.extractInputField(pd, ['$..path']);
+    for (const path of paths) {
+      this.modifyPathsWithSpecialCharacter(path.path, pd);
+    }
+  }
+
+  private static modifyPathsWithSpecialCharacter(pathDetails: (string | number)[], pd: IPresentationDefinition) {
+    let objectCursor: any = pd;
+    for (let i = 1; i < pathDetails.length; i++) {
+      if (i + 1 < pathDetails.length) {
+        objectCursor = objectCursor[pathDetails[i]];
+      }
+      if (pathDetails.length == i + 1) {
+        const paths: string[] = objectCursor[pathDetails[i]];
+        const editedPaths: string[] = [];
+        for (let j = 0; j < paths.length; j++) {
+          editedPaths.push(this.modifyPathWithSpecialCharacter(paths[j]));
+        }
+        objectCursor[pathDetails[i]] = editedPaths;
+        break;
+      }
+    }
+  }
+
+  private static modifyPathWithSpecialCharacter(path: string): string {
```

Comment:
A shorter implementation that has a better regex as it takes into account the closing bracket and also removes superfluous code:

```
 private static modifyPathWithSpecialCharacter(path: string): string {
    const REGEX_PATH = /(?<!\[')(@\w+)(?!\w'])/g;
    const matches = path.matchAll(REGEX_PATH);
    let next = matches.next();
    while (!next.done) {
      path = path.replace(next.value[1], ".['" + next.value[1] + "']");
      next = matches.next();
    }
    return path;
  }
```